tests/unit/OfficeAddresses.spec.ts

```diff
@@ -1,555 +1,1137 @@
+
+    // verify header
+    {
+      const header = editRecordsAddress.find('.address-edit-header .address-edit-title')
+      expect(header.text()).toBe('Records Office')
+    }
+
+    // verify delivery address
+    {
+      const block = editRecordsAddress.findAll('li').at(1)
+      expect(block.find('label').text()).toBe('Delivery Address')
+      expect(block.find('.inherit-checkbox').props('inputValue')).toBe(true)
+      expect(block.find('#address-records-delivery').exists()).toBe(false)
+    }
+
+    wrapper.destroy()
   })
 })
 
-describe('should properly emit valid - BCOMP', () => {
-  let wrapper: any
-  const localVue = createLocalVue()
-  const invalidDeliveryAddress = {
-    addressCity: 'someCity',
-    addressCountry: 'someCountry',
-    addressRegion: 'someRegion',
-    postalCode: 'somePostalCode',
-    streetAddress: 'someStreet'
-  }
+describe('"same as" checkboxes', () => {
+  let wrapper: any = null
 
-  const invalidMailingAddress = {
-    addressCity: 'someCity',
-    addressCountry: 'someCountry',
-    addressRegion: 'someRegion',
-    postalCode: 'somePostalCode',
-    streetAddress: 'someStreet'
-  }
+  beforeAll(() => {
+    // init entity type
+    store.state.stateModel.tombstone.entityType = 'BEN'
 
-  const validDeliveryAddress = {
-    addressCity: 'someCity1',
-    addressCountry: 'CA',
-    addressRegion: 'BC',
-    postalCode: 'somePostalCode',
-    streetAddress: 'someStreet'
-  }
+    // init original offices
+    store.state.stateModel.originalIA.incorporationApplication.offices = getIncorporationAddress(1, 1, 1, 1)
+  })
 
-  const validMailingAddress = {
-    addressCity: 'someCity2',
-    addressCountry: 'CA',
-    addressRegion: 'BC',
-    postalCode: 'somePostalCode',
-    streetAddress: 'someStreet'
-  }
+  beforeEach(async () => {
+    wrapper = mount(OfficeAddresses, { store, vuetify })
+
+    // set office addresses to trigger watcher
+    wrapper.vm.$store.commit('mutateOfficeAddresses', getIncorporationAddress(1, 1, 1, 1))
+    await Vue.nextTick()
+
+    // change to edit mode
+    const correctBtn = wrapper.find('#btn-correct-office-addresses')
+    correctBtn.trigger('click')
+    await Vue.nextTick()
+  })
 
   afterEach(() => {
     wrapper.destroy()
   })
 
+  it('toggles registered delivery address same as registered mailing address', async () => {
+    // verify initial mailing and delivery address data
+    expect(wrapper.vm.$data.mailingAddress.addressCity).toBe('addressCity1')
+    expect(wrapper.vm.$data.deliveryAddress.addressCity).toBe('addressCity1')
+
+    // verify that checkbox is checked and that address doesn't exist
+    const checkbox = wrapper.find('#registered-mailing-same-chkbx')
+    expect(checkbox.attributes('aria-checked')).toBe('true')
+    let address = wrapper.find('#address-registered-delivery')
+    expect(address.exists()).toBe(false)
+
+    // uncheck and verify the checkbox
+    checkbox.trigger('click')
```

Comment:
So, funny thing here (not haha :/ )...

If you 'find' a checkbox by its class then you get Vue's div for it (which contains the input element) and upon which you can call setChecked(bool). OTOH, if you 'find' a checkbox by its id then you get the input element and cannot call setChecked() on it. So I went with the cleanest code here.